crates/utf8-converter/README.md

```diff
@@ -0,0 +1,12 @@
+# UTF-8 Converter
+
+This crate converts string positions between Rust style (UTF-8 byte offsets) and styles used by other programming languages, as well as line numbers.
```

Comment:
Can you expand this a bit further?
(or do you plan to write more in a subsequent PR)

E.g. that all (most) operations are constant after some linear preprocessing of the input text.
And that space consumption is x bytes per input byte.

Reply:
Also there was: https://github.com/aneubeck/algos/blob/main/crates/utf8-converter/README.md
I.e. Greg prepared the move ~1 year ago (forgot about that one). Not sure how much it differs from our internal version.

Reply:
I'll steal Greg's text, and briefly say how to think about the performance.

I'll also link to [Tim's blog post](https://adaptivepatchwork.com/2023/07/10/utf-conversion/) about this.

Reply:
I'm going to make those changes in github/blackbird#8230, though, and update this PR when that's all merged.